inferno/trainers/basic.py

```diff
@@ -1365,9 +1366,11 @@ def apply_model_and_loss(self, inputs, target, backward=True, mode=None):
                 'trainer' in signature(self.criterion.forward).parameters):
             kwargs['trainer'] = self
         if mode == 'train':
-            loss = self.criterion(prediction, target, **kwargs)
+            loss = self.criterion(prediction, target, **kwargs) \
+                    if target else self.criterion(prediction, **kwargs) 
```

Comment:
I don't think this line is valid. `if target` will fail with `RuntimeError: bool value of Tensor with more than one value is ambiguous` if `target` has more than one element. If I understand correctly, you want to check if target has elements here. I think you can use [numel](https://pytorch.org/docs/stable/tensors.html#torch.Tensor.numel) for this.